test/unit/operations/invoke_host_function_test.js

```diff
@@ -128,6 +128,57 @@ describe('Operation', function () {
         expect(decodedOp.func.wasm()).to.eql(wasm);
         expect(decodedOp.auth).to.be.empty;
       });
+
+      it('lets you create contracts with a constructor', function () {
+        const h = hash(Buffer.from('random stuff'));
+        const constructorArgs = [
+          // note: using a string here doesn't work because once the operation
+          // is encoded/decoded it will be a Buffer internally and it'd be a
+          // mild pain in the ass to check equivalence
+          nativeToScVal(Buffer.from('admin name')),
```

Comment:
understood, but, it may be worthwhile to demonstrate plain string usage here as it represents typical client code scenario(and test is used as a reference by devs and followed verbatim), and then add the clunky buffer conversion down later for just test assertion purposes.

Reply:
fair enough :) way to advocate for the user :sunglasses: done in 8450ac2!